src/megaapi_impl.cpp

```diff
@@ -13943,29 +13943,41 @@ void MegaApiImpl::putnodes_result(const Error& inputErr, targettype_t t, vector<
             // folders for the new backup have been created -> now proceed with the sync
             handle backupHandle = nn.back().mAddedHandle;
             request->setNodeHandle(backupHandle);
-            std::unique_ptr<char[]> remotePath{ getNodePathByNodeHandle(nn.back().mAddedHandle) };
-            auto nextSyncTag = client->nextSyncTag();
+            std::unique_ptr<char[]> remotePath{ getNodePathByNodeHandle(backupHandle) };
 
             // create the SyncConfig
             const char* backupName = request->getName();
             const char* localPath = request->getFile();
-            SyncConfig syncConfig{ nextSyncTag, localPath, backupName, backupHandle, remotePath.get(),
-                                    0, {}, true, SyncConfig::TYPE_BACKUP };
+            auto regExp = request->getRegExp(); // not currently in use, but ready for addition
+            SyncConfig syncConfig{ localPath, backupName, backupHandle, remotePath.get(),
+                                    0, regExpToVector(regExp), true, SyncConfig::TYPE_BACKUP };
 
-            // add the Sync           
-            UnifiedSync* unifiedSync = nullptr;
-            e = client->addsync(syncConfig, DEBRISFOLDER, NULL, true, unifiedSync, false);
-            request->setNumDetails(syncConfig.getError());
-            if (!e)
+            client->addsync(syncConfig, DEBRISFOLDER, NULL, true, false,
+                                [this, request](UnifiedSync *unifiedSync, const SyncError &syncError, error e)
             {
-                unique_ptr<MegaSyncPrivate> sync(new MegaSyncPrivate(unifiedSync->mConfig, unifiedSync->mSync.get()));
-                request->setNumber(sync->getLocalFingerprint());
-                request->setTransferTag(nextSyncTag);
+                request->setNumDetails(syncError);
+
+                if (!e && !unifiedSync)
+                {
+                    e = API_ENOENT;
+                }
 
-                fireOnSyncAdded(sync.get(), e ? MegaSync::NEW_TEMP_DISABLED : MegaSync::NEW);
+                if (unifiedSync)
+                {
+                    if (unifiedSync->mSync)
+                    {
+                        fsfp_t fsfp = unifiedSync->mSync->fsfp;
+                        request->setNumber(fsfp);
+                    }
+                    request->setParentHandle(unifiedSync->mConfig.getBackupId());
+
+                    auto sync = ::mega::make_unique<MegaSyncPrivate>(unifiedSync->mConfig, unifiedSync->mSync.get());
+
+                    fireOnSyncAdded(sync.get(), e ? MegaSync::NEW_TEMP_DISABLED : MegaSync::NEW);
+                }
                 fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
-                return;
-            }
+            });
+
```

Comment:
If there's no return here, then the `fireOnRequestFinish()` (see [here](https://github.com/meganz/sdk/pull/2462/files#diff-5d048e8562b9eef0c31b18c647b312f4abf03481607422ffb7f1b12814524fc0R13984)) will delete the `request` and the lambda will crash at its call to `fireOnRequestFinish()`. I'll fix it.

Reply:
Done here: https://github.com/meganz/sdk/pull/2462/commits/a8c0dff752a9a3e823e658f3b2d519a181b76674